SFTP upload step fails strict host-key check in upload-benchmark-superset.yml workflow

**Describe the bug**
upload-benchmark-superset.yml is a workflow_call-triggered job with no actions/checkout step. It runs on a fresh runner, so ${GITHUB_WORKSPACE} is empty when the SFTP upload step points -oUserKnownHostsFile at .github/public_keys/umd-microbench-sftp-known-hosts.

**Environment**
- main CI build-and-run-all-benchmarks workflow

**Logs**
Run command -v sftp >/dev/null 2>&1 || { sudo apt-get update && sudo apt-get install -y openssh-client; }
  command -v sftp >/dev/null 2>&1 || { sudo apt-get update && sudo apt-get install -y openssh-client; }
  if ! compgen -G "${EXPORT_DIR}/*.json" > /dev/null; then
    echo "No benchmark-export JSON files found to upload." >> "${GITHUB_STEP_SUMMARY}"
    exit 0
  fi
  KEY_FILE="$(mktemp)"
  BATCH="$(mktemp)"
  trap 'rm -f "${KEY_FILE}" "${BATCH}"' EXIT
  printf '%s\n' "${SFTP_KEY}" > "${KEY_FILE}"
  chmod 600 "${KEY_FILE}"

  for f in "${EXPORT_DIR}"/*.json; do
    printf 'put "%s"\n' "${f}" >> "${BATCH}"
  done
  printf 'ls -l\n' >> "${BATCH}"
  sftp -oStrictHostKeyChecking=yes \
    -oUserKnownHostsFile="${GITHUB_WORKSPACE}/.github/public_keys/umd-microbench-sftp-known-hosts" \
    -i "${KEY_FILE}" -b "${BATCH}" "${SFTP_USER}@${SFTP_HOST}"
  shell: /usr/bin/bash -e {0}
  env:
    EXPORT_DIR: /tmp/superset-export
    SFTP_USER: umd-microbench-writer
    SFTP_HOST: s-dbd4b8a190fa40a4b.server.transfer.us-east-2.amazonaws.com
    SFTP_KEY: ***
  
No RSA host key is known for s-dbd4b8a190fa40a4b.server.transfer.us-east-2.amazonaws.com and you have requested strict checking.
Host key verification failed.
Connection closed
Error: Process completed with exit code 255.

**To reproduce**
CI run.

**Expected behavior**
Successful json upload over sftp.

--- a/.github/workflows/upload-benchmark-superset.yml
+++ b/.github/workflows/upload-benchmark-superset.yml
@@ -36,6 +36,13 @@ jobs:
       actions: read
 
     steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github/public_keys/umd-microbench-sftp-known-hosts
+          sparse-checkout-cone-mode: false
+
       - name: Download all Superset export artifacts
         uses: actions/download-artifact@v8
         with:

